pkg/util/validation/errors.go

```diff
@@ -0,0 +1,191 @@
+package validation
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/prometheus/common/model"
+	"github.com/weaveworks/common/httpgrpc"
+
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+)
+
+// ValidationError is an error returned by series validation. It provides a function to convert
+// the error into an HTTPGRPC error.
+//
+// nolint:golint ignore stutter warning
+type ValidationError interface {
+	error
+
+	// ToHTTPGRPCError returns the httpgrpc version of the error.
+	ToHTTPGRPCError() error
+}
+
+// genericValidationError is a basic implementation of ValidationError which can be used when the
+// error format only contains the cause and the series.
+type genericValidationError struct {
+	message string
+	cause   string
```

Comment:
This naming confused me... `cause` is simply the first formatting-argument for `message`.

Reply:
I see your point but I can't get a better idea for naming, so I just used `cause` which was the variable name already used in `ValidateLabels()`.